community-tutorials/how-to-install-and-configure-webhook-to-trigger-a-shell-script/01-en.md

```diff
@@ -0,0 +1,97 @@
+---
+title: How to install and configure a webook to trigger a shell script
+description: This tutorial describes how to install a webhook from adnanh. Webhook is a service written in go which wil expose a lightweight webserver. If you call a specific route on the webserver you can execute a command or a shell script on your server.
+updated_at: 2021-11-06
+slug: how-to-install-and-configure-webhook-to-trigger-a-shell-script
+author_name: Patrick Weber
+author_url: -
+author_image: -
+author_bio: -
+tags: debian, ubuntu
+netcup_product_url: https://www.netcup.de/vserver/vps.php
+language: en
+available_languages: en
+---
+
+# Introduction
+This tutotiral is about how to install and configure a webhook from adnanh to trigger a shell script.
+In the shell script you can start a deployment or some other actions.
+
+# Requirements
+First you need the package manager apt. You need also an editor, for example vi or nano, to create and change config files.
+
+# Step 1 - Updatating your packages and install webhook from adnanh
+```
+sudo apt-get update && sudo apt-get -y upgrade
+sudo apt-get install webhook
+```
+
+Then check the version: ```webhook -version```
+
+# Step 2 - Create the webhook-configuration file
+Create the webhook-configuration file with an editor: ```nano /etc/webhook.conf```
+
+Paste the webhook-configuration:
+```
+[
+   {
+      "id": "deploy-webhook",
+      "execute-command": "/var/app/webhook-entrypoint.sh",
+      "command-working-directory": "/var/app",
+      "include-command-output-in-response": true,
+      "include-command-output-in-response-on-error": true,
+   }
+]
+```
+Save the configuration file.
+
+We defined only one webhook in an array in the json-config-file. Here we are using the execute-command property to define the shell-script-path. In the command-working-directory the base path is defined.
+Later in this tutorial we will execute our webhook with a http request.
+With the property "include-command-output-in-response" webhook will include the stdoutput of the shellscript to the http response body. With the property "include-command-output-in-response-on-error" webhook include the erroutput of the shellscript to the http response body. The http-request will wait until the full output is collected. If the exit-code of the shell-script is zero, the status-code of a http-request is 200. If the exit-code is not zero, the status-code is not 200.
+You have the option to remove the two "include-*" properties, then you will not see the output of the shellscript in the http-response body. For security reasons, its recommended to not include the output in the http-response-body.
+
+# Step 3 - Create the app-directory
+Create the app directory: ```mkdir -p /var/app```
+
+# Step 4 - Create the webhook-entrypoint.sh and make it executeable
+Create the file ```/var/app/webhook-entrypoint.sh``` with the content:
+
```

Comment:
You describe how I create a folder. good

Then I have to write a content in a file that is in this folder.  
**But how do I create the file?**

Reply:
done, thanks a lot